pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -210,14 +214,58 @@ func (r *ReconcileZookeeperCluster) reconcileStatefulSet(instance *zookeeperv1be
 		}
 		return r.updateStatefulSet(instance, foundSts, sts)
 	}
-	return nil
 }
 
 func (r *ReconcileZookeeperCluster) updateStatefulSet(instance *zookeeperv1beta1.ZookeeperCluster, foundSts *appsv1.StatefulSet, sts *appsv1.StatefulSet) (err error) {
 	r.log.Info("Updating StatefulSet",
 		"StatefulSet.Namespace", foundSts.Namespace,
 		"StatefulSet.Name", foundSts.Name)
 	zk.SyncStatefulSet(foundSts, sts)
+
+	//Getting the upgradeCondition from the zk clustercondition
+	_, upgradeCondition := instance.Status.GetClusterCondition(zookeeperv1beta1.ClusterConditionUpgrading)
+
+	//Setting the upgrade condition to true when the zk cluster is upgrading
+	//When the zk cluster is upgrading currentversion is not empty as well as Statefulset CurrentRevision and UpdateRevision are not equal and zk cluster image tag is not equal to CurrentVersion
+	if upgradeCondition.Status != corev1.ConditionTrue && instance.Status.CurrentVersion != "" && foundSts.Status.CurrentRevision != foundSts.Status.UpdateRevision && instance.Spec.Image.Tag != instance.Status.CurrentVersion {
```

Comment:
Could we also check whether PodsReady is true before setting upgrade condition to true?

Reply:
Also before referring to `upgradeCondition.Status` we should do a nil check on upgradeCondition

Reply:
I have added both.